doc/generate-kuadrant-auth-policy.md

```diff
@@ -41,21 +45,17 @@ components:
       openIdConnectUrl: https://example.com/.well-known/openid-configuration
```

Comment:
This particular example is also broken and we should not be sharing it without at least a similar warning to the user as [this one proposed further below](https://github.com/Kuadrant/kuadrantctl/blob/2efa953196af640783868c100a7c115f4e8d2b46/doc/generate-kuadrant-auth-policy.md?plain=1#L318).

_Explanation:_

Copying a well-known [OpenId Connect Discovery](https://openid.net/specs/openid-connect-discovery-1_0.html) URL to the AuthPolicy's `jwt.issuerUrl` field will not work.

That's why kuadrantctl actually expects in the `openIdConnectUrl` field of the OAS an _issuer URL_ identifier instead. This is a URL without the `/.well-known/openid-configuration` part, which is [automatically appended](https://github.com/coreos/go-oidc/blob/22dfdcabd450013b4d51ac15b6423f529d957e9f/oidc/oidc.go#L209) to the value obtained from `jwt.issuerUrl`.

Users who follow the [OAS docs](https://swagger.io/docs/specification/authentication/openid-connect-discovery/) though may inadvertently use the full Discovery URL. We do not want that.

Reply:
This is a good catch! Please look at my next commit a9db0d01de625302a56b2b64eb2ef28054268e99
I removed the mention of the `/.well-known/openid-configuration` path so it should be less confusing.

I am just wondering if there is a reason why AuthPolicy expects `issuer` instead of `openid-configuration` path? 
The response JSON from `openid-configuration` contains value of `issuer` so theoretically it could be used as well. Also if the current practice is to use `openid-configuration` path in OAS yaml-s as you mentioned.

Reply:
> I am just wondering if there is a reason why AuthPolicy expects `issuer` instead of `openid-configuration` path?

It's because of the OIDC library used by Authorino. In its simplest form, it expects an issuer URL and not the (more specific) Discovery endpoint. Otherwise, Authorino itself would have to perform the request to fetch the OIDC config, parse its response, handle possible exceptions, and then start using the library that, among other things, let us validate signed JWTs.

I would advocate for more flexibility in Authorino's [JWT verification method](https://docs.kuadrant.io/authorino/docs/features/#jwt-verification-authenticationjwt), which surfaces into the Kuadrant AuthPolicy. Along with `issuerUrl`, I'd love to add `discoveryUrl`, `jwksURL`, and possibly others. It's just that currently this is not the state of things... yet.